### ISSUE TITLE
The RSI calculation currently uses a simple moving average (SMA) for computing average gains and losses. Using an exponentially weighted moving average (EWMA) can provide smoother and more responsive RSI values, which is the standard approach in most trading platforms.
Suggested change:
avg_gain = gain.ewm(span=window, adjust=False).mean()
avg_loss = loss.ewm(span=window, adjust=False).mean()

### ISSUE THREAD

**Issue by abbyturi** [UBC-MDS/traders_copilot_mzza_25#57]

**Comment by abbyturi** [comment]
We acknowledge that this package is a simplified implementation, primarily designed as a foundational framework. With this in place, while EWMA offers smoother and more responsive RSI values, it also introduces a lag and may overemphasize recent price changes, which might not always align with all trading strategies.   Future advancements should incorporate more comprehensive financial metrics for portfolio composition and investment decision-making, including a broader range of technical indicators, market sentiment analysis, and real-time data integration. We leave these aspects open as directions for future work.
